### DISCUSSION TITLE
How to map between AWS private IP address and public IP address?

### DISCUSSION THREAD

**Discussion by smulikHakipod** [siderolabs/talos#9125] [General]
Hey, I am trying to create a control plane between the same region AWS / Hetnzer. 
Hetnzer are mapping the public IP to the VM interface, which is great - talos is able to determine the right IP address. 
AWS are mapping virtual private IPs to Public IPs, meaning talos using the private IP. It means talos is falling to join the AWS machine as its trying to contact its private IP instead of Public. 

Is there a way to tell talos to use the public IP in AWS and not the 172.x.x.x IP when advertising its IP to rest of the cluster?

**Comment by smira** [comment]
I'm not quite sure what you're asking about, if you run AWS image of Talos, it learns its public IP from AWS metadata service. 

It also depends on what exactly you're trying to advertise, and what fails exactly.

**Reply by smulikHakipod** [reply]
Thanks @smira 
I am attaching a screenshot from AWS instance running Talos. 
As you can see, the IP its seeing is 172.x.x.x - its private (there is a public IP assigned to the machine). 


<img width="853" alt="Screenshot 2024-08-07 at 19 47 39" src="https://github.com/user-attachments/assets/8096e32e-b599-4e0b-8d3a-8261af906d78">

While  `talosctl --talosconfig talosconfig   get affiliates` show the private IP of the AWS but the Public IP of hetzner - so Hetnzer and AWS wont be able to communicate ever with each other. 

Is Talos suppose to see the public IP of AWS in that case? If so - why its using the private IP?

Thanks

**Reply by smira** [reply]
when connecting across the clouds, the easiest is to enable [KubeSpan](https://www.talos.dev/v1.7/talos-guides/network/kubespan/).

**Reply by smulikHakipod** [reply]
@smira Thanks, that actually worked, I now can see both nodes as control planes in the same cluster 
```
 hetnzer-packer % KUBECONFIG=./kubeconfig kubectl get nodes      
NAME                    STATUS   ROLES           AGE     VERSION
ip-172-31-1-242         Ready    control-plane   7m21s   v1.30.3
talos-control-plane-2   Ready    control-plane   7m15s   v1.30.3
```
